pkg/datarepair/checker/checker.go

```diff
@@ -217,3 +170,120 @@ func contains(a []string, x string) bool {
 	}
 	return false
 }
+
+func (checker *Checker) checkSegmentStatus(ctx context.Context, pointer *pb.Pointer, path string, monStats *durabilityStats) (err error) {
+	remote := pointer.GetRemote()
+	if remote == nil {
+		return nil
+	}
+
+	pieces := remote.GetRemotePieces()
+	if pieces == nil {
+		checker.logger.Debug("no pieces on remote segment")
+		return nil
+	}
+
+	missingPieces, err := checker.overlay.GetMissingPieces(ctx, pieces)
+	if err != nil {
+		checker.logger.Error("error getting missing pieces: ", zap.Error(err))
+		return err
+	}
+
+	monStats.remoteSegmentsChecked++
+	pathElements := storj.SplitPath(path)
+	if len(pathElements) >= 2 && pathElements[1] == "l" {
+		monStats.remoteFilesChecked++
+	}
+
+	numHealthy := int32(len(pieces) - len(missingPieces))
+	redundancy := pointer.Remote.Redundancy
+	// we repair when the number of healthy files is less than or equal to the repair threshold
+	// except for the case when the repair and success thresholds are the same (a case usually seen during testing)
+	if numHealthy >= redundancy.MinReq && numHealthy <= redundancy.RepairThreshold && redundancy.RepairThreshold != redundancy.SuccessThreshold {
+		if len(missingPieces) == 0 {
+			checker.logger.Warn("Missing pieces is zero in checker, but this should be impossible -- bad redundancy scheme.")
+			return nil
+		}
+		monStats.remoteSegmentsNeedingRepair++
+		err = checker.repairQueue.Insert(ctx, &pb.InjuredSegment{
+			Path:       path,
+			LostPieces: missingPieces,
+		})
+		if err != nil {
+			checker.logger.Error("error adding injured segment to queue: ", zap.Error(err))
+			return err
+		}
+
+		// if same segment previously was irreparable
+		if _, err = checker.irrdb.Get(ctx, []byte(path)); err == nil {
```

Comment:
When method `checkSegmentStatus` is triggered from `irreparableProcess()` then you are sure that this element was from `irrdb`. Can you change logic to avoid asking `irrdb` in such case to reduce DB traffic?

Reply:
irreparableProcess() runs off of irrdb and needs to read each entry and run the checksegmentstatus(). not sure how can I get pass the segment info. The irreparable process()'s reads the pointers data from irrdb and not the meta info db.

Reply:
done